apps/moose-console/src/components/query-interface.tsx

```diff
@@ -0,0 +1,285 @@
+    "TRIGGER",
+    "PROCEDURE",
+    "FUNCTION",
+    "PACKAGE",
+    "LOCK",
+    "EXPLAIN",
+    "DESCRIBE",
+    "ANALYZE",
+    "GRANT",
+    "REVOKE",
+    "PRIVILEGES",
+    "COMMIT",
+    "ROLLBACK",
+    "SAVEPOINT",
+    "SET",
+    "SESSION",
+    "SYSTEM",
+    "USER",
+    "DATABASE",
+    "SCHEMA",
+    "DOMAIN",
+    "TYPE",
+    "CHARACTER",
+    "COLLATION",
+    "TRANSLATION",
+    "SERVER",
+    "CONNECTION",
+    "STATEMENT",
+    "PREPARE",
+    "EXECUTE",
+    "DEALLOCATE",
+    "WORK",
+    "ISOLATION",
+    "LEVEL",
+    "READ",
+    "WRITE",
+    "ONLY",
+    "CALL",
+    "RETURN",
+    "HANDLER",
+    "CONDITION",
+    "SIGNAL",
+    "RESIGNAL",
+    "ITERATE",
+    "LEAVE",
+    "LOOP",
+    "REPEAT",
+    "UNTIL",
+    "OPEN",
+    "CLOSE",
+    "FETCH",
+    "DECLARE",
+    "CURSOR",
+    "CONTINUE",
+    "EXIT",
+    "GET",
+    "DIAGNOSTICS",
+    "STACKED",
+    "DYNAMIC",
+    "STATIC",
+    "SENSITIVE",
+    "PRIOR",
+    "SQLSTATE",
+    "SQLCODE",
+    "SQLERROR",
+    "SQLWARNING",
+    "SQLNOTFOUND",
+    "SQLROWCOUNT",
+    "SQLFOUND",
+    "SQL",
+]
+
+async function runQuery(queryString: string): Promise<any> {
+    const client = getClient();
+  
+    const resultSet = await client.query({
+      query: queryString,
+      format: "JSONEachRow",
+    });
+  
+    return resultSet.json();
+  }
+
+
+interface QueryInterfaceProps {
+    table: Table,
+    related: Table[]
+}
+
+const insertSomeText = (insert: string, originalValue: string, ref: MutableRefObject<HTMLTextAreaElement>, setter: Dispatch<SetStateAction<string>>) => {
+    const selectionStart = ref.current.selectionStart;
+    const selectionEnd = ref.current.selectionEnd;
+ 
+    let newValue =
+      originalValue.substring(0, selectionStart) +
+      insert +
+      originalValue.substring(selectionEnd, originalValue.length);
+    setter(newValue);
+ };
+ 
+
+export default function QueryInterface({ table, related }: QueryInterfaceProps) {
+    // Create a ref to the textarea
+    const textareaRef = useRef<HTMLTextAreaElement>(null);
+    const tables = related.filter(t => !t.name.includes(".inner")).map(t => `${t.database}.${t.name}`)
+
+    const[value, setValue] = useState(`SELECT * FROM ${table.database}.${table.name} LIMIT 50;`)
+    const [results, setResults] = useState<any[]>()
+    const [sqlKeyWordCount, setSqlKeyWordCount] = useState(12)
+    const [tableCount, setTableCount] = useState(12)
+
+    useEffect(() => {
+        (
+            async () => {
+                const data = await runQuery(value)
+                setResults(data)
+            }
+        )()
+        
+        const handleKeyDown = (e: KeyboardEvent) => {
+            if (e.key === "Enter" && (e.ctrlKey || e.metaKey)) {
+                runQuery(value).then((results) => {
+                    e.preventDefault()
+                    setResults(results)
+                })
+            }
+        }
+
+        document.addEventListener("keydown", handleKeyDown)
```

Comment:
Hum should you be adding event listener on the whole document? feels like that should only be on the text area no?

Reply:
No it's for the run run shortcut. Whole doc is the way to go I believe. The removal happens when the component goes out of view